client-lib/src/main/java/com/ibm/ws/repository/connections/RepositoryConnection.java

```diff
@@ -89,8 +90,17 @@
     public Collection<EsaResource> getMatchingEsas(FilterableAttribute attribute, String identifier) throws RepositoryBackendException;
 
     /**
-     * This method gets all the resources in this repository
+     * This method will return all resources that match the supplied filter logic
      * 
+     * @param predicate The filters + logic to apply
+     * @return The resources of any type that match the filters
+     * @throws RepositoryBackendException
+     */
+    public Collection<RepositoryResource> getMatchingResources(FilterPredicate... predicate) throws RepositoryBackendException;
```

Comment:
I'm unsure about the name `getMatchingResources`

"Matching" is already an overloaded term within the client-lib, I'd prefer not to overload it further.

Would `getFilteredResources` be better?